src/include/ipc/header.h

```diff
@@ -217,6 +233,19 @@ struct sof_ipc_compound_hdr {
 	uint32_t count;		/**< count of 0 means end of compound sequence */
 } __attribute__((packed));
 
+/**
+ * Flexible array header
+ *
+ * Commands that contain flexible arrays as the only member in payload should
+ * add this as secondary header, right after main IPC header, to provide count
+ * of elements in following flex-size array.
+ */
+struct sof_ipc_flexible_array_hdr {
+	uint32_t nelements;	/**< Count of elements in flexible array
+				 *   following this header in payload
+				 */
+};
+
```

Comment:
It's weird, better just make it one struct with count  and array

Reply:
@jajanusz If we had templates, that would be a way to go. But since we don't know the type of array members up front, in general, it doesn't make much sense. Putting placeholder `uint8_t[0]` would introduce confusion, as whether it should denote size of the array in bytes, or different, intended member types.
So, as a resolution to conflicting requirements (keeping cavs interface intact and denoting count of array members), this solution prepends payload with a header with count of array elements alone.